opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/authenticate/embedded/Xs2aAisAuthenticateUserConsentWithPin.java

```diff
@@ -50,7 +50,9 @@ protected void doValidate(DelegateExecution execution, Xs2aContext context) {
     @Override
     protected void doRealExecution(DelegateExecution execution, Xs2aContext context) {
         logResolver.log("doRealExecution: execution ({}) with context ({})", execution, context);
-
+        if (context.isEmbeddedPreAuthDone()) {
```

Comment:
This check should be removed from here and instead one should use it in diagram exclusive gateway